deltacat/exceptions.py

```diff
@@ -1,14 +1,145 @@
-class RetryableError(Exception):
-    pass
+from __future__ import annotations
+from enum import Enum
 
 
-class NonRetryableError(Exception):
-    pass
```

Comment:
Why removed this? It will be easy to retry on all retryable errors using tenacity.